### ISSUE TITLE
How do I get all the VALID tickers for Yahoo Finance?

### ISSUE THREAD

**Issue by kabonker** [ranaroussi/yfinance#402]
I keep getting errors on some tickers, and Is there a quick way to see all the tickers they support?

Is there a function to call?   Or a page to go to?

thanks
john

**Comment by ScriptKittyM30wM30w** [comment]
@kabonker I have same question, currently using symbols scraped from [EODData](https://eoddata.com/symbols.aspx) . You can select the exchange you want and pull all tickers from each page of the alphabet (all of this is selectable from URL so easily scrapable). I haven't run the whole list yet but will post it once I do. Can post scraper code to GitHub in meantime and update this comment with link.

**Comment by kabonker** [comment]
I created a list of about 4000 tickers.. but would be cool if we just had a list in this library that is some how synced with yahoo finance, so everyone doesn't have to do it manually themselves.

**Comment by ScriptKittyM30wM30w** [comment]
Agreed, need to compile the list first. May not be publicly available anywhere on Yahoo finance site though so brute force is only solution. BTW can you post your list?

**Comment by ScriptKittyM30wM30w** [comment]
@kabonker is this the source you used?
http://www.nasdaqtrader.com/trader.aspx?id=symboldirdefs

**Comment by Gunjan933** [comment]
@kabonker I found out **[this website](https://investexcel.net/all-yahoo-finance-stock-tickers/)** which has all the tickers [last updated 2017]. So you can use that.

#### This Excel spreadsheet was updated in September 2017.  It is broken down into eight sheets, one each for:
* stocks (106332 tickers)
* ETFs (21196tickers)
* futures (9294 tickers)
* indices (80017 tickers)
* mutual funds (24926 tickers)
* currency pairs (4019 tickers)

#### That’s 245784 ticker symbols in total.

**Comment by kabonker** [comment]
Dude that is awesome...    Scotty we will need more power!!!!   https://youtu.be/QYqXSlPmMJs?t=10

**Comment by taewookim** [comment]
FYI this file says "2017".. been 3 years. Must be missing a ton of companies. Anyone have an updated list?

**Comment by beglitis** [comment]
I am interested in this one as well. One other problem is that I get a lot of tickers as not existing/delisted/deleted. This is obviously an issue. If a ticker was live in 2017 but the company has been delisted since then and therefore the ticker is not valid then there is no consistent way of creating accurate historic stock universes for backtesting. To make this a bit dramatic, Enron would not appear in any of the universes because it doesn't exist today however if you were doing live trading you would have obviously included it at the time (and taken the hit if you were long). In a technical jargon, the survivorship bias is massive. I guess this is why it pays to have professional grade data. That said, this is better than nothing for small projects.

**Comment by DimitriBolt** [comment]
Does someone know a python-module that can obtain the ticker list?

**Comment by kabonker** [comment]
Maybe someone can just write a simple program going through the alphabet
for 4 chars  and make a list of all the successful calls, then write it
out..

that would be nice.   Could just run it once a quarter..   like COIN would
be new.   Then spit out the full list and the incremental so we know the
new ones..

**Comment by Fethe98** [comment]
Information from https://finviz.com/screener.ashx?ft=4 and package from https://pypi.org/project/pyfinviz/.

See the code below. (You can add more than 10 pages if you want all the tickers...)

Have a nice day! :)

`from pyfinviz.screener import Screener

page = 10

screener = Screener(view_option=Screener.ViewOption.VALUATION,
                     pages=[x for x in range(1, page)])


list_ticker = []
for i in range(0, page):
    if i == 1:
        pass
    else:
        for j in range(len(screener.data_frames[i])):
            list_ticker.append(screener.data_frames[i].Ticker[j])
            list(list_ticker)

print(list_ticker)`

**Comment by shullaw** [comment]
How many should there be?  I can scrape 10,000 when looking for ALL tickers, just not sure if I got them all because they claim to have 90,000+

**Comment by atanasi99** [comment]
Hey everyone, any updates on retrieving all tickers from Yahoo finance. Also does anyone know any good APIs for retrieving FX options data. Thanks

**Comment by vishalbelsare** [comment]
https://github.com/mlapenna7/yh_symbol_universe

**Comment by Mxttak** [comment]
You can use the screeners available at the Yahoo finance site: https://finance.yahoo.com/screener/ 
For example this screener gives you all ETFs traded in the US https://finance.yahoo.com/screener/unsaved/5719e91c-77b9-42ff-987b-d99bcf5c9c8d (3429 on 2022-12-29)

**Comment by AGDholo** [comment]
How about https://[api.nasdaq.com/api/screener/stocks?tableonly=true&download=true](https://api.nasdaq.com/api/screener/stocks?tableonly=true&download=true)
